src/test/scala/akka/persistence/jdbc/query/HardDeleteQueryTest.scala

```diff
@@ -0,0 +1,103 @@
+package akka.persistence.jdbc.query
+
+import akka.persistence.query.{ EventEnvelope, NoOffset, Sequence }
+import akka.pattern._
+import com.typesafe.config.ConfigFactory
+import org.scalactic.source.Position
+import org.scalatest.Matchers
+
+import scala.concurrent.duration._
+
+abstract class HardDeleteQueryTest(config: String) extends QueryTestSpec(config) with Matchers {
+
+  implicit val askTimeout = 500.millis
+
+  it should "not return deleted events when using CurrentEventsByTag" in withActorSystem { implicit system =>
+    val journalOps = new ScalaJdbcReadJournalOperations(system)
+    withTestActors(replyToMessages = true) { (actor1, _, _) =>
+      (actor1 ? withTags(1, "number")).futureValue
+      (actor1 ? withTags(2, "number")).futureValue
+      (actor1 ? withTags(3, "number")).futureValue
+
+      // delete all three events and wait for confirmations
+      (actor1 ? DeleteCmd(1)).futureValue shouldBe "deleted-1"
+      (actor1 ? DeleteCmd(2)).futureValue shouldBe "deleted-2"
+      (actor1 ? DeleteCmd(3)).futureValue shouldBe "deleted-3"
```

Comment:
I improve the tests for the hard delete. 

We must delete all events and check that nothing is returned. As such, we prove that the last event, the one that stays as logically deleted, is not returned from the query.

We don't need to test that there is still one logically deleted event in the journal because the test-kit captures that already. That's how we discovered that we need to keep the last one.